src/components/tree/phyloTree/helpers.js

```diff
@@ -78,27 +81,78 @@ function _getSpaceBetweenSubtrees(numSubtrees, numTips) {
  * rectangularLayout, radialLayout, createChildrenAndParents
  * side effects: <phyloNode>.displayOrder (i.e. in the redux node) and <phyloNode>.displayOrderRange
  * @param {Array<PhyloNode>} nodes
+ * @param {boolean} focus
  * @returns {undefined}
  */
-export const setDisplayOrder = (nodes) => {
+export const setDisplayOrder = (nodes, focus) => {
+  timerStart("setDisplayOrder");
+
+  const getDisplayOrder = getDisplayOrderCallback(nodes, focus);
   const numSubtrees = nodes[0].n.children.filter((n) => n.fullTipCount!==0).length;
   const numTips = nodes[0].n.fullTipCount;
   const spaceBetweenSubtrees = _getSpaceBetweenSubtrees(numSubtrees, numTips);
   let yCounter = 0;
   /* iterate through each subtree, and add padding between each */
   for (const subtree of nodes[0].n.children) {
     if (subtree.fullTipCount===0) { // don't use screen space for this subtree
-      setDisplayOrderRecursively(nodes[subtree.arrayIdx], undefined);
+      setDisplayOrderRecursively(nodes[subtree.arrayIdx], getDisplayOrder, undefined);
     } else {
-      yCounter = setDisplayOrderRecursively(nodes[subtree.arrayIdx], yCounter);
+      yCounter = setDisplayOrderRecursively(nodes[subtree.arrayIdx], getDisplayOrder, yCounter);
       yCounter+=spaceBetweenSubtrees;
     }
   }
   /* note that nodes[0] is a dummy node holding each subtree */
   nodes[0].displayOrder = undefined;
   nodes[0].displayOrderRange = [undefined, undefined];
+
+  timerEnd("setDisplayOrder");
 };
 
+/**
+ * @param {Array<PhyloNode>} nodes
+ * @param {boolean} focus
+ * @returns fn to return a display order (y position) for a node
+ */
+function getDisplayOrderCallback(nodes, focus) {
+  /**
+   * Start at 0 and increase with each node.
+   * Note that this value is shared across invocations of the callback.
+   */
+  let displayOrder = 0;
+
+  /**
+   * Keep track of whether the previous node was selected
+   */
+  let previousWasVisible;
+
+  if (focus) {
+    const numVisible = nodes.filter((d) => !d.hasChildren && d.visibility === NODE_VISIBLE).length;
```

Comment:
🐛  
```diff
-     const numVisible = nodes.filter((d) => !d.hasChildren && d.visibility === NODE_VISIBLE).length;
+    const numVisible = nodes.filter((d) => !d.n.hasChildren && d.visibility === NODE_VISIBLE).length;
```

But even simpler, just use `nodes[0].n.tipCount`

Reply:
> I plan to circle back to it after this PR!

Nice one. Feel free to effect `/tipCount/visibleTipCount/` at that stage if you'd like.